firefox-ios/Client/Frontend/Browser/Tabs/Middleware/TabManagerMiddleware.swift

```diff
@@ -899,7 +899,12 @@ class TabManagerMiddleware {
     private func addToBookmarks(_ shareItem: ShareItem?) {
         guard let shareItem else { return }
         // Add new mobile bookmark at the top of the list
-        profile.places.createBookmark(parentGUID: BookmarkRoots.MobileFolderGUID,
+        let isBookmarksRefactorEnabled = featureFlags.isFeatureEnabled(.bookmarksRefactor, checking: .buildOnly)
+        let bookmarksSaveToFolderGuid = profile.prefs.stringForKey(PrefsKeys.BookmarkSaveToFolder)
+        let parentGuid = isBookmarksRefactorEnabled ?
+        bookmarksSaveToFolderGuid ?? BookmarkRoots.MobileFolderGUID : BookmarkRoots.MobileFolderGUID
```

Comment:
It might just be the lack of oxygen to my brain this week but this took me awhile to realized that line 905 was attached to the logic of 904. Instead of going `bookmarksSaveToFolderGuid ?? BookmarkRoots.MobileFolderGUID` could we just add 
`let bookmarksSaveToFolderGuid = profile.prefs.stringForKey(PrefsKeys.BookmarkSaveToFolder)` to the guard statement so we can shorten everything on those two lines?

Reply:
some indentation would have probably helped 😅 we can't add that to the guard statement because we want to proceed with creating the bookmark, regardless of if the pref is set. 

What do you think about this:
`let parentGuid = (isBookmarkRefactorEnabled ? recentBookmarkFolderGuid : nil) ?? BookmarkRoots.MobileFolderGUID`

It's a bit unconventional, but I think it's easier to read and less redundant (doesn't use `BookmarkRoots.MobileFolderGUID` twice)